.github/workflows/branch.yml

```diff
@@ -0,0 +1,17 @@
+name: Branch
+
+on:
+  push:
+    branches-ignore:
+      - dev
```

Comment:
Are we *not* testing on a push to dev?

Reply:
We could remove the reservation, but the "main" branch, or dev as it's being used here, and gets its own workflow, e.g. `.github/workflows/dev.yml`, to include logic like publishing to NPM. Then we reserve the branch workflow away from dev so they don't compete with each other.

Reply:
OK, good

Reply:
I added logic to a separate job in the workflow that checks if `github.ref == refs/heads/dev` so that we don't need two separate workflows.